ui/cypress/integration/common/volume.js

```diff
@@ -11,68 +11,98 @@ And('I choose the Volumes tag', () => {
-    cy.get('.sc-select__menu')
-      .find('[data-cy="type-sparseLoopDevice"]')
-      .click();
-
-    cy.get('input[name=sizeInput]').type(volumeCapacity);
-
-    cy.get('.sc-select').eq(2).click();
-    //cy.get('[data-cy="size-KiB"]').click(); leave it at the default value GiB
-
-    cy.get('[data-cy="submit-create-volume"]').click();
-
-    // Wait until the volume is ready
-    cy.waitUntil(
-      () =>
-        cy
-          .get('[data-cy=volume_status_value]')
-          .then(($span) => $span.text() === 'Ready'),
-      {
-        errorMsg: `Volume ${volumeNameSparseLoopDevice} is not ready`,
-        timeout: 120000, // waits up to 120000 ms, default to 5000
-        interval: 5000, // performs the check every 5000 ms, default to 200
-      },
-    );
-
-    // Check if all the labels are presents
-    cy.get('[data-cy=volume_label_name]').should('contain', volume_label_name);
-    cy.get('[data-cy=volume_label_value]').should(
-      'contain',
-      volume_label_value,
-    );
-  },
-);
+And('I click [Create] button', () => {
+  cy.get('[data-cy="submit-create-volume"]').click();
+});
+
+Then('I am redirected to the "test-volume-sparse" volume page', () => {
+  cy.location('pathname').should('eq', '/volumes/test-volume-sparse');
+});
+
+And('the volume "test-volume-sparse" becomes Ready', () => {
+  // Check if the right side panel updates
+  cy.get('[data-cy="volume_detail_card_name"]').should(
+    'contain',
+    'test-volume-sparse',
+  );
+  // Wait until the volume is ready
+  cy.waitUntil(
+    () =>
+      cy
+        .get('[data-cy=volume_status_value]')
+        .then(($span) => $span.text() === 'Ready'),
+    {
+      errorMsg: `Volume test-volume-sparse is not ready`,
+      timeout: 120000, // waits up to 120000 ms, default to 5000
+      interval: 5000, // performs the check every 5000 ms, default to 200
+    },
+  );
+});
+
+And('the label of volume "test-volume-sparse" presents:', (dataTable) => {
+  const dataTableObject = dataTable.rowsHash();
+
+  const volumeLabelName = dataTableObject.labelName;
+  const volumeLabelValue = dataTableObject.labelValue;
+
+  cy.get('[data-cy="volume_label_name"]').should('contain', volumeLabelName);
+  cy.get('[data-cy="volume_label_value"]').should('contain', volumeLabelValue);
+});
+
+When('I click [Delete] button', () => {
+  cy.get('[data-cy="delete_volume_button"]').click({ force: true });
+});
+
+And('I confirm the deletion', () => {
+  cy.get('[data-cy="confirm_deletion_button"]').click();
+});
+
+Then('the "test-volume-sparse" volume is removed from the list', () => {
+  // fiter delete this volume, we should only have 3 default volume in CI.
+  cy.waitUntil(
+    () =>
+      cy.get('[data-cy="volume_table_row"]').then(($el) => $el.length === 3),
+    {
+      errorMsg: `Volume test-volume-sparse is not deleted`,
+      timeout: 120000, // waits up to 120000 ms, default to 5000
+      interval: 5000, // performs the check every 5000 ms, default to 200
+    },
+  );
```

Comment:
You should be able to search through the found elements here:

```javascript
  cy.waitUntil(
    () =>
      cy.get('[data-cy="volume_table_row"]').then(($rows) => {
        $rows.forEach(($row) => {
          nameCell = // some query on $row children
          expect(nameCell).to.not.contain('test-volume-sparse')
        )
      }),
    {
      errorMsg: `Volume test-volume-sparse is not deleted`,
      timeout: 120000, // waits up to 120000 ms, default to 5000
      interval: 5000, // performs the check every 5000 ms, default to 200
    },
  );
```

But, more simply (to be tested), it seems the [`contains`](https://docs.cypress.io/api/commands/contains.html) command can search deep in the DOM, so this would probably be the simplest:

```suggestion
  cy.waitUntil(
    () =>
      cy.contains('[data-cy="volume_table_row"]', 'test-volume-sparse')
      	.should('not.exist'),
    {
      errorMsg: `Volume test-volume-sparse is not deleted`,
      timeout: 120000, // waits up to 120000 ms, default to 5000
      interval: 5000, // performs the check every 5000 ms, default to 200
    },
  );
```

Reply:
Hello @gdemonet 
Note that: you cannot put assertions inside `checkFunction` (https://github.com/NoriSte/cypress-wait-until)
> In Cypress, a failing assertion forces the whole test to fail and you cannot restore it

I am afraid neither of them could work, so I didn't find other solutions than check there are 3 volume in the table

Reply:
Ah, right, waiting is built-in many Cypress commands actually, so you don't need `waitUntil` at all: https://docs.cypress.io/api/commands/should.html#Wait-until-the-assertions-pass